ui/src/dialogs/AddToPlaylistDialog.js

```diff
@@ -44,34 +46,37 @@ export const AddToPlaylistDialog = () => {
 
   const addToPlaylist = (playlistId, distinctIds) => {
     const trackIds = Array.isArray(distinctIds) ? distinctIds : selectedIds
-    dataProvider
-      .create('playlistTrack', {
-        data: { ids: trackIds },
-        filter: { playlist_id: playlistId },
-      })
-      .then(() => {
-        const len = trackIds.length
-        notify('message.songsAddedToPlaylist', 'info', { smart_count: len })
-        onSuccess && onSuccess(value, len)
-        refresh()
-      })
-      .catch(() => {
-        notify('ra.page.error', 'warning')
-      })
+    if (trackIds.length) {
+      dataProvider
+        .create('playlistTrack', {
+          data: { ids: trackIds },
+          filter: { playlist_id: playlistId },
+        })
+        .then(() => {
+          const len = trackIds.length
+          notify('message.songsAddedToPlaylist', 'info', { smart_count: len })
+          onSuccess && onSuccess(value, len)
+          refresh()
+        })
+        .catch(() => {
+          notify('ra.page.error', 'warning')
+        })
+    } else {
+      notify('message.songsAddedToPlaylist', 'info', { smart_count: 0 })
+    }
   }
 
   const checkDuplicateSong = (playlistObject) => {
-    dataProvider
-      .getOne('playlist', { id: playlistObject.id })
+    httpClient(`${REST_URL}/playlist/${playlistObject.id}/tracks`)
```

Comment:
Is there a reason for switching from `dataProvider` to `httpClient`? Ideally we want to use `dataProvider` as much as possible. You can access a playlist's track by using the `playlistTrack` resource (with `playlist_id` filter). Example: https://github.com/navidrome/navidrome/blob/d3e142233b3f072df834c9f048996a2cdc783e47/ui/src/playlist/PlaylistActions.js#L45-L50

Reply:
I tried to do it that way, but ran into the issue of that messing with the track list on display. If I'm in a playlist and try to add a song from that playlist to another one, when I click on the name of the other playlist, the `dataprovider` updates the list in memory and causes a change to the list displayed.

You can see what I mean in this gif
![dataprovider](https://user-images.githubusercontent.com/24394068/193074713-6f84e274-b556-47ce-a928-31de74a5609d.gif)

Reply:
Yeah, you are right. I wonder how this worked before... :/

Anyway, can you please fix the tests so I can merge this one? Thanks!